src/plugins/oauth2/openIdConfiguration.ts

```diff
@@ -1,26 +1,63 @@
 import { log, userInteractionProvider } from '../../io';
 import type * as models from '../../models';
 import * as utils from '../../utils';
+import { isEqualWith } from 'lodash';
 import get from 'lodash/get';
 import { URL } from 'url';
 
 export const DEFAULT_CALLBACK_URI = 'http://localhost:3000/callback';
 const DefaultOAuthVariablePrefix = 'oauth2';
 
-function getVariable(variables: models.Variables, variablePrefix: string | undefined, name: string): string {
-  const getter = (prefix: string) => variables[`${prefix}_${name}`] || get(variables, `${prefix}.${name}`);
+function getVariableRaw(variables: models.Variables, name: string, variablePrefix: string): unknown {
+  let varValue = variables[`${variablePrefix}_${name}`];
+  if (typeof varValue === 'undefined' || varValue === null) {
+    varValue = get(variables, `${variablePrefix}.${name}`);
+  }
+  return varValue;
+}
+
+function getVariableUnknown(variables: models.Variables, variablePrefix: string | undefined, name: string): unknown {
+  const getter = getVariableRaw.bind(undefined, variables, name);
   let value: unknown;
   if (variablePrefix) {
     value = getter(variablePrefix);
   }
-  if (!value) {
+  if (value === undefined) {
     value = getter(DefaultOAuthVariablePrefix);
   }
   const expandedValue = utils.expandVariable(value, variables);
+  return expandedValue;
+}
+
+function getString(variables: models.Variables, variablePrefix: string | undefined, name: string): string {
+  const expandedValue = getVariableUnknown(variables, variablePrefix, name);
+  if (typeof expandedValue === 'undefined' || expandedValue === null) {
+    return '';
+  }
   if (utils.isString(expandedValue)) {
     return expandedValue;
   }
-  return '';
+  return `${expandedValue}`;
+}
+
+function getBooleanLike(
+  variables: models.Variables,
+  variablePrefix: string | undefined,
+  name: string,
+  defaultValue = false
+): boolean {
+  const expandedValue = getVariableUnknown(variables, variablePrefix, name);
+  if (typeof expandedValue === 'boolean') return expandedValue;
+  if (typeof expandedValue === 'number') return !!expandedValue;
+  if (typeof expandedValue === 'undefined' || expandedValue === null) return defaultValue;
+  const stringValue = utils.isString(expandedValue) ? expandedValue : `${expandedValue}`;
```

Comment:
Why is `getString` not used here? This would also provide a correct undefined handling.

Reply:
Commit [3f1f8da](https://github.com/AnWeber/httpyac/pull/228/commits/3f1f8da865981ea27ecc63258eeb29bb28cd2dcb)